standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -4693,6 +4694,17 @@ private Table ensureGetTable(String catName, String dbName, String tblName)
     return convertToTable(ensureGetMTable(catName, dbName, tblName));
   }
 
+  /**
+   * Conditionally alters an {@link ExpressionTree} to be compatible with certain database products.
+   */
+  private void reconcileTreeWithDatabaseProduct(ExpressionTree tree) throws MetaException {
+    if (sqlGenerator.getDbProduct().isDERBY()) {
```

Comment:
Is this only problem for Derby?

Reply:
@findepi mentioned in https://issues.apache.org/jira/browse/HIVE-21614?focusedCommentId=16849571&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-16849571 that Oracle may be implicated as well, but I have not verified.

Having this `ObjectStore#reconcileTreeWithDatabaseProduct` defined allows us to conveniently add other databases, though, so perhaps we could revisit as needed?

Reply:
There are Docker images for the supported databases. Could you please run tests with them?